airbyte-cdk/python/airbyte_cdk/sources/file_based/config/abstract_file_based_spec.py

```diff
@@ -46,12 +47,18 @@ def schema(cls, *args: Any, **kwargs: Any) -> Dict[str, Any]:
         Generates the mapping comprised of the config fields
         """
         schema = super().schema(*args, **kwargs)
-        transformed_schema = copy.deepcopy(schema)
+        transformed_schema: Dict[str, Any] = copy.deepcopy(schema)
         schema_helpers.expand_refs(transformed_schema)
         cls.replace_enum_allOf_and_anyOf(transformed_schema)
+        cls.remove_discriminator(transformed_schema)
 
         return transformed_schema
 
+    @staticmethod
+    def remove_discriminator(schema: Dict[str, Any]) -> None:
+        """pydantic adds "discriminator" to the schema for oneOfs, which is not treated right by the platform as we inline all references"""
+        dpath.util.delete(schema, "properties/**/discriminator")
```

Comment:
should we do this at a higher level so it's applied to all connectors? maybe this belongs in `Stream.get_json_schema`?

Reply:
Probably - I will do this refactoring in a separate PR.